src/main/java/mate/controller/cars/AddCarController.java

```diff
@@ -0,0 +1,36 @@
+package mate.controller.cars;
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Car;
+import mate.service.CarService;
+import mate.service.ManufacturerService;
+
+@WebServlet(urlPatterns = "/cars/add", loadOnStartup = 1)
+public class AddCarController extends HttpServlet {
+    private static final Injector injector
+            = Injector.getInstance("mate");
+    private static final CarService carService
+            = (CarService) injector.getInstance(CarService.class);
+    private static final ManufacturerService manufacturerService
+            = (ManufacturerService) injector.getInstance(ManufacturerService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Car car = new Car();
+        String model = req.getParameter("carModel");
+        car.setModel(model);
+        car.setManufacturer(manufacturerService
+                .get(Long.parseLong(req.getParameter("manufacturerId"))));
+        if (!model.isEmpty()) {
+            carService.create(car);
+        }
```

Comment:
Do we really need this checking? If so, I think it is better to move validation at the begin of the method, before you have created an object

Reply:
I added this check to prevent adding car without model. But now i think that it will be better to move this check to service layer. Am i right?

Reply:
I don't think that any kind of validation is necessary in this task at all. But if you really would like to check if some input values is null, it is better to do at the begin of the method before creating an object and send to the user a corresponding message. Now it is not so important, anyway in a future you will use specific libraries for this